modules/data.atmosphere/R/tdm_temporal_downscale_functions.R

```diff
@@ -0,0 +1,1134 @@
+        mod.out <- parallel::mclapply(dat.list, model.train, mc.cores = n.cores, 
+            n.beta = n.beta, resids = resids, threshold = quantile(dat.train[dat.train$surface_downwelling_shortwave_flux_in_air > 
+                0, "surface_downwelling_shortwave_flux_in_air"], 0.05))
+        
+        # Use a loop to sace each day of year independently
+        for (i in names(mod.out)) {
+            # Save the betas as .nc
+            outfile <- file.path(path.out, paste0("betas_surface_downwelling_shortwave_flux_in_air_", 
+                i, ".nc"))
+            dimY <- ncdf4::ncdim_def(paste0("coeffs_", i), units = "unitless", 
+                longname = "model.out coefficients", vals = 1:ncol(mod.out[[i]][["betas"]]))
+            dimX <- ncdf4::ncdim_def("random", units = "unitless", longname = "random betas", 
+                vals = 1:nrow(mod.out[[i]][["betas"]]))
+            var.list <- ncdf4::ncvar_def(i, units = "coefficients", dim = list(dimX, 
+                dimY), longname = paste0("day ", i, " model.out coefficients"))
+            nc <- ncdf4::nc_create(outfile, var.list)
+            ncdf4::ncvar_put(nc, var.list, mod.out[[i]][["betas"]])
+            ncdf4::nc_close(nc)
+            
+            # Save the model as a .Rdata
+            mod.save <- mod.out[[i]][["model"]]
+            save(mod.save, file = file.path(path.out, paste0("model_surface_downwelling_shortwave_flux_in_air_", 
+                i, ".Rdata")))
+        }
+        
+    } else {
+        for (i in names(dat.list)) {
+            mod.out <- model.train(dat.subset = dat.list[[i]], threshold = quantile(dat.train[dat.train$surface_downwelling_shortwave_flux_in_air > 
+                0, "surface_downwelling_shortwave_flux_in_air"], 0.05), 
+                n.beta, resids = resids)
+            
+            # Save the betas as .nc
+            outfile <- file.path(path.out, paste0("betas_surface_downwelling_shortwave_flux_in_air_", 
+                i, ".nc"))
+            dimY <- ncdf4::ncdim_def(paste0("coeffs_", i), units = "unitless", 
+                longname = "model.out coefficients", vals = 1:ncol(mod.out[["betas"]]))
+            dimX <- ncdf4::ncdim_def("random", units = "unitless", longname = "random betas", 
+                vals = 1:nrow(mod.out[["betas"]]))
+            var.list <- ncdf4::ncvar_def(i, units = "coefficients", dim = list(dimX, 
+                dimY), longname = paste0("day ", i, " model.out coefficients"))
+            nc <- ncdf4::nc_create(outfile, var.list)
+            ncdf4::ncvar_put(nc, var.list, mod.out[["betas"]])
+            ncdf4::nc_close(nc)
+            
+            # Save the model as a .Rdata
+            mod.save <- mod.out$mode
+            save(mod.save, file = file.path(path.out, paste0("model_surface_downwelling_shortwave_flux_in_air_", 
+                i, ".Rdata")))
+        }
+    }
+    
+    return(mod.out)
+}
```

Comment:
This function looks A LOT like the previous function -- seems like there should be a way to either make one master function (which is passed a variable specific list of covariates and linear model syntax) and has a couple special case if's, rather than having a lot of different functions. At a minimum, it looks like there are large chunks that could be pulled out to other functions so you can call then every time.